File: frontend/src/services/subscription.ts
import api from './api'

export interface SubscriptionPlan {
  id: string
  name: string
  tier: string
  billing_cycle: string
  price: number
  currency: string
  features: string[]
}

export interface CreateSubscriptionData {
  plan_name: string
  plan_tier: string
  billing_cycle: 'monthly' | 'annual'
  price: number
  currency?: string
}

export interface Subscription {
  id: number
  company_id: number
  plan_name: string
  plan_tier: string
  status: string
  billing_cycle: string
  price: number
  currency: string
  starts_at: string
  ends_at: string | null
}

export const subscriptionService = {
  async getAvailablePlans(): Promise<{ plans: SubscriptionPlan[] }> {
    const response = await api.get<{ plans: SubscriptionPlan[] }>('/api/subscription/plans')
    return response.data
  },

  async createSubscription(data: CreateSubscriptionData): Promise<Subscription> {
    const response = await api.post<Subscription>('/api/subscription', data)
    return response.data
  }
}

File: backend/app/models/subscription.py
from sqlalchemy import Column, Integer, String, ForeignKey, Enum, Numeric, DateTime
from sqlalchemy.orm import declarative_base, relationship
from datetime import datetime
import enum
from app.models.person import Base

class SubscriptionStatus(str, enum.Enum):
    TRIAL = "trial"
    ACTIVE = "active"
    CANCELLED = "cancelled"
    EXPIRED = "expired"

class BillingCycle(str, enum.Enum):
    MONTHLY = "monthly"
    ANNUAL = "annual"

class Subscription(Base):
    __tablename__ = "subscriptions"
    
    id = Column(Integer, primary_key=True, index=True)
    company_id = Column(Integer, ForeignKey("companies.id"), nullable=False, index=True)
    plan_name = Column(String, nullable=False)
    plan_tier = Column(String, nullable=False)
    status = Column(Enum(SubscriptionStatus), default=SubscriptionStatus.TRIAL, nullable=False, index=True)
    billing_cycle = Column(Enum(BillingCycle), nullable=False)
    price = Column(Numeric(10, 2), nullable=False)
    currency = Column(String, default="USD", nullable=False)
    starts_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    ends_at = Column(DateTime, nullable=True)
    cancelled_at = Column(DateTime, nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)

